### DISCUSSION TITLE
Issue: Consistent error with installation (singularity)

### DISCUSSION THREAD

**Discussion by lkearney001** [aertslab/scenicplus#122] [General]
When building a Docker image the following error occurs. Please assist if possible.

-- lawrence

podman build -t aertslab/scenicplus:latest . -f scenicplus/Dockerfile

Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [4 lines of output]
      Requirement already satisfied: numpy>=1.17 in /tmp/pip-build-env-mw6zttkb/overlay/lib/python3.8/site-packages (1.24.2)
      error in MACS2 setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Expected end or semicolon (after name and no valid version specifier)
          numpy>=>=1.17
               ^
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
Error: error building at STEP "RUN pip install --no-cache-dir --upgrade pip wheel &&     pip install --no-cache-dir numpy==1.22 &&     pip install --no-cache-dir MACS2 &&     pip install --no-cache-dir scipy==1.8.0 &&     pip install --no-cache-dir Cython &&     pip install --no-cache-dir fitsne &&     pip install --no-cache-dir papermill &&     pip install --no-cache-dir igv_jupyterlab &&     pip install --no-cache-dir bs4 &&     pip install --no-cache-dir lxml &&     pip install --no-cache-dir tspex &&     pip install --no-cache-dir plotly &&     pip install --no-cache-dir kaleido &&     pip install --no-cache-dir pyvis &&     pip install velocyto &&     pip install --no-cache-dir pygam &&     pip install --no-cache-dir fa2 &&     pip install --no-cache-dir scanpy==1.8.2 &&     pip install --no-cache-dir -r /tmp/requirements.txt": error while running runtime: exit status 1

**Comment by lkearney001** [comment]
Still hoping I can get some help with this one, really need to deploy this as a singularity container.


-- lawrence

**Comment by SeppeDeWinter** [comment]
Hi @lkearney001 

Your issue sounds similar to this one: https://github.com/aertslab/scenicplus/issues/101

The solution I gave there might be able to help you.

Best,

Seppe

**Comment by lkearney001** [comment]
Seppe,
Yes, it does sound similar, I did see it, but how is can it be applied to the singularity container build issue when the install set is referenced in the docker repo (aertslab/scenicplus:latest)?

-- lawrence

**Reply by pajucon** [reply]
Remove the original pip install MACS line from the dockerfile.
Substitute with a pip install from the local corrected MACS directory (reference the pip local installs lower in the dockerfile).

**Reply by hangarbait** [reply]
... ty vry much. I'll test and advise of the result.


-- lawrence
